contrib/partitioner/partitioner.py

```diff
@@ -107,12 +119,24 @@ def main():
                 sys.exit(1);
         else:
             print("Already partitioned")
-
-
+    
 
         if len(usb_partitions()) == 1:
             try:
                 os.system('/bin/mount /dev/' + usb_partitions()[0] + ' /mnt/data')
+                
+                # If not EXT4, delete partition and format
+                if not os.path.exists('/mnt/data/lost+found'):
+                    print('Not an EXT filesystem, remove all partitions')
+                    partitions = [];
+                    for p in usb_partitions():
+                        partitions.append(p.replace('sda',''));
+                    for pn in partitions:
+                        os.system('parted -s /dev/sda rm ' + pn);
+                    print('Running parted, and recreating partition as EXT4')
+                    os.system('/sbin/parted -s /dev/sda mkpart p ext4 3 100%');
+                    os.system('/sbin/mkfs.ext4 -F /dev/sda1');
```

Comment:
the first drive is always sda on rpi. Never had it mount as sdb before.

Unless you have a better idea.

Reply:
I actually had it as sdb the other day. It was the only drive plugged in.

I did do something weird, I think I reformatted while it was still mounted. After boot is was back to sda again. Doesn't seem very likely to occur but doesn't seem like we should trust it.

Hardcoding to sda is probably safe if we do it everywhere, but not if we do it some places and iterate over `/sys/block/*` elsewhere like we are now. Since then we might test stuff on sdb and try to perform actions on sda.

Reply:
I removed all instances of sda in the code